seml/config.py

```diff
@@ -327,9 +327,38 @@ def convert_values(val):
     return val
 
 
+class YamlUniqueLoader(yaml.FullLoader):
+    """
+    Custom YAML loader that disallows duplicate keys
+
+    From https://github.com/encukou/naucse_render/commit/658197ed142fec2fe31574f1ff24d1ff6d268797
+    Workaround for PyYAML issue: https://github.com/yaml/pyyaml/issues/165
+    This disables some uses of YAML merge (`<<`)
+    """
+
+
+def construct_mapping(loader, node, deep=False):
+    """Construct a YAML mapping node, avoiding duplicates"""
+    loader.flatten_mapping(node)
```

Comment:
This only works on the uppermost level, right?

E.g. works, doesn't it?
```
fixed:
  a.b: ccc
  a.b: ddd
```

Reply:
No, this will work on any level. In your example this will detect that `a.b` is defined twice.

If you instead define `a: {'b': 'ddd'}`, that duplicate will not be caught by this code but by something Daniel has implemented earlier.

Reply:
Right, of course.